public/app/features/dashboard-scene/panel-edit/PanelEditorRenderer.tsx

```diff
@@ -1,61 +1,72 @@
 import { css } from '@emotion/css';
-import React from 'react';
+import React, { useMemo } from 'react';
 
 import { GrafanaTheme2 } from '@grafana/data';
 import { SceneComponentProps } from '@grafana/scenes';
-import { Splitter, useStyles2 } from '@grafana/ui';
+import { Splitter, ToolbarButton, useStyles2 } from '@grafana/ui';
 
 import { NavToolbarActions } from '../scene/NavToolbarActions';
 import { getDashboardSceneFor } from '../utils/utils';
 
 import { PanelEditor } from './PanelEditor';
-import { VisualizationButton } from './PanelOptionsPane';
 
 export function PanelEditorRenderer({ model }: SceneComponentProps<PanelEditor>) {
   const dashboard = getDashboardSceneFor(model);
-  const { optionsPane, vizManager, dataPane } = model.useState();
+  const { optionsPane, vizManager, dataPane, optionsPaneSize } = model.useState();
   const { controls } = dashboard.useState();
   const styles = useStyles2(getStyles);
+  const [vizPaneStyles, optionsPaneStyles] = useMemo(() => {
+    if (optionsPaneSize > 0) {
+      return [{ flexGrow: 1 - optionsPaneSize }, { minWidth: 'unset', overflow: 'hidden', flexGrow: optionsPaneSize }];
+    } else {
+      return [{ flexGrow: 1 }, { minWidth: 'unset', flexGrow: 0 }];
+    }
+  }, [optionsPaneSize]);
 
   return (
     <>
       <NavToolbarActions dashboard={dashboard} />
-      <div className={styles.canvasContent}>
-        {controls && (
-          <div className={styles.controls}>
-            {controls.map((control) => (
-              <control.Component key={control.state.key} model={control} />
-            ))}
-            {!optionsPane && (
-              <VisualizationButton
-                pluginId={vizManager.state.panel.state.pluginId}
-                onOpen={() => model.toggleOptionsPane(true)}
-                isOpen={false}
-                onTogglePane={() => model.toggleOptionsPane()}
-              />
-            )}
-          </div>
-        )}
+      <Splitter
+        direction="row"
+        dragPosition="end"
+        initialSize={0.75}
+        primaryPaneStyles={vizPaneStyles}
+        secondaryPaneStyles={optionsPaneStyles}
+        onResizing={model.onOptionsPaneResizing}
+        onSizeChanged={model.onOptionsPaneSizeChanged}
+      >
         <div className={styles.body}>
-          <Splitter
-            direction="row"
-            dragPosition="end"
-            initialSize={0.75}
-            primaryPaneStyles={{ paddingBottom: !dataPane ? 16 : 0 }}
-          >
+          <div className={styles.canvasContent}>
+            {controls && (
```

Comment:
This can be a bit confusing because controls are coming from dashboard, but we'll have other controls in here available only in panel edit mode (stuff like Table view,  Fill/Acutal etc.) This wasn't an issue in the old arch because the time picker was in the top navbar. 

Maybe we can leave this as it is for now and just update the dashboard controls prop with items that should show up in panel edit mode. But in future we should probably separate dashboard and panel controls.